components/HackathonModal.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from 'react';
+import Modal from 'react-modal';
+import Image from '../components/Image';
+import Link from 'next/link';
+import styles from '../styles/components/HackathonModal.module.scss';
+import { website_config } from '../config';
+
+const MODAL_DISPLAY_INTERVAL = 2 * 24 * 60 * 60 * 1000; // 2 days in milliseconds
+
+interface HackathonModalProps {
+  onRequestClose: () => void;
+}
+
+const HackathonModal: React.FC<HackathonModalProps> = ({ onRequestClose }) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [assetsLoaded, setAssetsLoaded] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fontFamilies = ['Orbitron', 'Poppins', 'Motorblock'];
+
+    const imageUrls = [
+      '/Hackathon/little-guy.png',
+      '/Hackathon/top-left.svg',
+      '/Hackathon/top-right.svg',
+      '/Hackathon/bottom-left.svg',
+    ];
+
+    const loadFonts = async () => {
+      try {
+        await Promise.all(
+          fontFamilies.map((family) => document.fonts.load(`1em "${family}"`))
+        );
+      } catch (error) {
+        console.error('Failed to load fonts:', error);
+      }
+    };
+
+    const loadImages = async () => {
```

Comment:
Instead of doing this what if you just did `import ImageName from 'svgfile'` next.js supports it and will inline them meaning we dont need to wait for the build (If these are the svgs wrapping pngs we should just use the raw data instead of putting them in a png and the <image element should handle this, we lose out on image optimization this way and its added compelxity`